src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -0,0 +1,221 @@
+    *  For instance:
+    *
+    *  ```scala
+    *  EmptyCatchBlock(tree).error(pos)   // res: Error
+    *  EmptyCatchBlock(tree).warning(pos) // res: Warning
+    *  ```
+    */
+  import dotc.ast.Trees._
+  import dotc.ast.untpd
+
+  // Syntax Errors ---------------------------------------------------------- //
+  abstract class EmptyCatchOrFinallyBlock(tryBody: untpd.Tree, errNo: Int)(implicit ctx: Context)
+  extends Message(errNo) {
+    val explanation = {
+      val tryString = tryBody match {
+        case Block(Nil, untpd.EmptyTree) => "{}"
+        case _ => tryBody.show
+      }
+
+      val code1 =
+        s"""|try $tryString catch {
+            |  case t: Throwable => ???
+            |}""".stripMargin
+
+      val code2 =
+        s"""|try $tryString finally {
+            |  // perform your cleanup here!
+            |}""".stripMargin
+
+      hl"""|A ${"try"} expression should be followed by some mechanism to handle any exceptions
+           |thrown. Typically a ${"catch"} expression follows the ${"try"} and pattern matches
+           |on any expected exceptions. For example:
+           |
+           |$code1
+           |
+           |It is also possible to follow a ${"try"} immediately by a ${"finally"} - letting the
+           |exception propagate - but still allowing for some clean up in ${"finally"}:
+           |
+           |$code2""".stripMargin
+    }
+  }
+
+  case class EmptyCatchBlock(tryBody: untpd.Tree)(implicit ctx: Context)
+  extends EmptyCatchOrFinallyBlock(tryBody, 1) {
+    val kind = "Syntax"
+    val msg =
+      hl"""|The ${"catch"} block does not contain a valid expression, try
+           |adding a case like - `${"case e: Exception =>"}` to the block""".stripMargin
+  }
+
+  case class EmptyCatchAndFinallyBlock(tryBody: untpd.Tree)(implicit ctx: Context)
+  extends EmptyCatchOrFinallyBlock(tryBody, 2) {
+    val kind = "Syntax"
+    val msg =
+      hl"""|A ${"try"} without ${"catch"} or ${"finally"} is equivalent to putting
+           |its body in a block; no exceptions are handled.""".stripMargin
+  }
+
+  case class DeprecatedWithOperator()(implicit ctx: Context)
+  extends Message(3) {
+    val kind = "Syntax"
+    val msg =
+      hl"""${"with"} as a type operator has been deprecated; use `&' instead"""
+    val explanation = {
+      val codeBlock1 =
+        """|trait A {
+           |  type T = Int
+           |}
+           |
+           |trait B {
+           |  type T = Double
+           |}""".stripMargin
+
+      hl"""|Dotty introduces intersection types - `&' types. These replace the
+           |use of the ${"with"} keyword. There are a few differences in
+           |semantics between intersection types and using `${"with"}'.
+           |
+           |`${"A with B"}' is ordered, `${"A & B"}' is not.
+           |
+           |In:
+           |
+           |$codeBlock1
+           |
+           |The type of `${"T"}' in `${"A with B"}' is ${"Int"} whereas in `${"A & B"}'
```

Comment:
This can be confusing since you're explaining the semantics of `A with B` in scalac and not in dotty where it's identical to  `A & B`, I think you should make the explanation shorter and refer to a wiki page with the full details that we should write.